packages/redis-cache/src/index.ts

```diff
@@ -9,25 +10,59 @@ export class RedisCache implements CacheStrategy {
     this.client.connect()
   }
 
-  async get(_pageKey: string, cacheKey: string): Promise<CacheEntry | null> {
-    const pageData = await this.client.get(cacheKey)
+  async get(pageKey: string, cacheKey: string): Promise<CacheEntry | null> {
+    const pageData = await this.client.get(`${pageKey}//${cacheKey}`)
     if (!pageData) return null
     return JSON.parse(pageData)
   }
 
-  async set(_pageKey: string, cacheKey: string, data: CacheEntry): Promise<void> {
-    await this.client.set(cacheKey, JSON.stringify(data))
+  async set(pageKey: string, cacheKey: string, data: CacheEntry): Promise<void> {
+    await this.client.set(`${pageKey}//${cacheKey}`, JSON.stringify(data), {
+      ...(data.revalidate ? { EX: Number(data.revalidate) } : undefined)
+    })
   }
 
-  async delete(_pageKey: string, cacheKey: string): Promise<void> {
-    await this.client.del(cacheKey)
+  async revalidateTag(tag: string): Promise<void> {
+    // Revalidate by Path
+    if (tag.startsWith(NEXT_CACHE_IMPLICIT_TAG_ID)) {
+      await this.deleteAllByKeyMatch(tag.slice(NEXT_CACHE_IMPLICIT_TAG_ID.length))
+      return
+    }
+
+    // Revalidate by Tag
+    let cursor = 0
+    do {
+      const { cursor: currentCursor, keys } = await this.client.scan(0, { MATCH: '*', COUNT: 100 })
```

Comment:
in redis we can query by value https://redis.io/docs/latest/develop/interact/search-and-query/query/, so possible we can fetch all data which contains specific tag inside

Reply:
This doesn't seem to be a quick change. Can I make this a separate ticket?